File: strongtify_client/src/types/artist.ts
import { Album } from "./album";
import { Song } from "./song";

export type Artist = {
    id: string;
    name: string;
    alias: string;
    imageUrl?: string;
    followerCount: number;
}

export type ArtistDetail = {
    id: string;
    createdAt: string;
    updatedAt: string;
    name: string;
    alias: string;
    birthDate?: string;
    about?: string;
    imageUrl?: string;
    followerCount: number;
    songCount: number;
    albumCount: number;
    songs?: Song[];
    albums?: Album[];
}

export type CudArtistResponse = {
    id: string;
    name: string;
    alias: string;
    imageUrl?: string;
    followerCount: number;
    birthDate?: Date;
    about?: string;
}

export type CreateUpdateArtistRequest = {
    name: string;
    birthDate?: string;
    about?: string;
    image?: File;
}

export type ArtistDetailQuery = {
    songLimit?: number;
    albumLimit?: number;
}

File: strongtify_mobile_app/lib/models/artist/artist.dart
class Artist {
  Artist({
    required this.id,
    required this.name,
    required this.alias,
    this.imageUrl,
    this.followerCount = 0,
  });

  final String id;
  final String name;
  final String alias;
  final String? imageUrl;
  final int followerCount;

  factory Artist.fromMap(Map<String, dynamic> data) {
    return Artist(
      id: data['id'],
      name: data['name'],
      alias: data['alias'],
      imageUrl: data['imageUrl'],
      followerCount: data['followerCount'] ?? 0,
    );
  }
}
